app/lib/green_lanes_updates_publisher/data_updates_finder.rb

```diff
@@ -0,0 +1,48 @@
+module GreenLanesUpdatesPublisher
+  class DataUpdatesFinder
+    DB = Sequel::Model.db
+
+    def initialize(date = Time.zone.today)
+      @start_date = date
+    end
+
+    def call
+      all_updates = []
+      updates = fetch_updates(base_regulations)
+      updates.each { |update| all_updates.push(update) }
+
+      updates = fetch_updates(modification_regulations)
+      updates.each { |update| all_updates.push(update) }
+      all_updates
+    end
+
+    private
+
+    def fetch_updates(query)
+      dataset = DB.fetch(query, @start_date, @start_date)
+      dataset.map { |row| GreenLanesUpdate.new(row[:regulation_id], row[:regulation_role], row[:measure_type_id]) }
+    end
+
+    def base_regulations
+      "select br.base_regulation_id as regulation_id, br.base_regulation_role as regulation_role, mo.measure_type_id
+      from base_regulations_oplog br
+      inner join measures_oplog mo
+      on br.base_regulation_id = mo.measure_generating_regulation_id and br.base_regulation_role = mo.measure_generating_regulation_role
+      inner join measure_types mt
+      on mo.measure_type_id = mt.measure_type_id and mt.trade_movement_code in (0,1)
+      where br.created_at > ? or mo.created_at > ?
+      group by(br.base_regulation_id, br.base_regulation_role, mo.measure_type_id)"
+    end
+
+    def modification_regulations
+      "select mr.modification_regulation_id as regulation_id, mr.modification_regulation_role as regulation_role, mo.measure_type_id
```

Comment:
Should we create active record queries for this or use arel tables instead of raw sql queries mainly for sql injection?

something like;

```

Copy code
class ModificationRegulation < ApplicationRecord
  has_many :measures, foreign_key: 'measure_generating_regulation_id', primary_key: 'modification_regulation_id'
end

class Measure < ApplicationRecord
  belongs_to :modification_regulation, foreign_key: 'measure_generating_regulation_id', primary_key: 'modification_regulation_id'
  belongs_to :measure_type, foreign_key: 'measure_type_id', primary_key: 'measure_type_id'
end

class MeasureType < ApplicationRecord
  has_many :measures, foreign_key: 'measure_type_id', primary_key: 'measure_type_id'
end

result = ModificationRegulation
  .joins(measures: :measure_type)
  .where('modification_regulations.created_at > ? OR measures.created_at > ?', time1, time2)
  .where(measure_types: { trade_movement_code: [0, 1] })
  .group('modification_regulations.modification_regulation_id', 'modification_regulations.modification_regulation_role', 'measures.measure_type_id')
  .select('modification_regulations.modification_regulation_id AS regulation_id',
          'modification_regulations.modification_regulation_role AS regulation_role',
          'measures.measure_type_id')
          ```

Reply:
Im not sure what our best practise is for something like this i know arel tables can look a little complex

Reply:
just had a look and arel tables might get bit long winded for this

Reply:
actually above might be wrong as not sure how the `measures_oplog` table affects